components/hab/src/main.rs

```diff
@@ -304,10 +305,38 @@ fn sub_pkg_export(ui: &mut UI, m: &ArgMatches) -> Result<()> {
 }
 
 fn sub_pkg_hash(m: &ArgMatches) -> Result<()> {
-    let source = m.value_of("SOURCE").unwrap(); // Required via clap
-
     init();
-    command::pkg::hash::start(&source)
+    match m.value_of("SOURCE") {
+        Some(source) => {
+            // hash single file
+            command::pkg::hash::start(&source)
+        },
+        None => {
+            // read files from stdin
+            let stdin = io::stdin();
+            for line in stdin.lock().lines() {
+                match line {
+                    Ok(line) => {
+                        let file = line.trim_right();
+                        if Path::new(&file).is_file() {
+                            match command::pkg::hash::start(&file) {
+                                Ok(_) => (),
+                                Err(error) => {
+                                    writeln!(io::stderr(), "{}: error hashing file: {}", &file, error).unwrap();
+                                }
+                            }
+                        } else {
+                            writeln!(io::stderr(), "{}: No such file or directory", &file).unwrap();
+                        }
+                    },
+                    Err(error) => {
+                        writeln!(io::stderr(), "error: {}", error).unwrap();
+                    },
```

Comment:
The `Err` match on line 332 is actually from the lines() iterator. I'm not sure if it's really needed or if we can just `line.unwrap()` instead.

The `Err` match on 324 is needed to log individual errors within a list of files, if there was an issue hashing one of them. Not really sure the set of possible errors here..